### ISSUE TITLE
Use constructor dependency injection for default component model

### ISSUE THREAD

**Issue by frecco75** [mapstruct/mapstruct#2211]
Hello,

On MapStruct 1.4.0, we can override the injection strategy using `mapstruct.defaultInjectionStrategy=constructor` but it doesn't work with `mapstruct.defaultComponentModel=default`.

I think it would be a great to have the possibility to use constructor IoC without Spring or CDI.

**Comment by filiphr** [comment]
The problem with that is that `Mappers.getMapper` will not work anymore if we do that and you will have to instantiate your mappers on your own. Perhaps this is something that we could do if you decide to use `constructor` injection strategy with the default component model.

**Comment by frecco75** [comment]
Yes it is not a problem to instantiate a mapper in a configuration file.
An example is for example an application built with the clean architecture, and using Spring framework. 

We could consider mapStruct as a jdk extension and use it in the domain module (which doesn't depend on Spring), and instantiate the mapper in a bootstrap/infra module in a Spring configuration file with a dedicated method annotate with @Bean.
